### PR TITLE
implement function result value not used inspection (resolves #203)

### LINKED ISSUE #203: Function return value is not used
Consider a function:

    Public Function Foo(ByVal bar As String) As Integer
        Foo = 42
    End Function

Then a function call:

        Foo "bar"

Either the function has side-effects and should be a procedure, or not using the returned value is a programming error. Rubberduck should warn about the function's return value not being used; formal definition would be something like "a function call that isn't on the right-hand side of a local variable assignment".

**Possible quick-fix:**
- [ ] Introduce variable: this introduces a local declaration on the line immediately above the function call, and an assignment of that variable with the function call on the right-hand side of the `=` operator.

### PR DESCRIPTION
**Pull request by autoboosh**

### CODE REVIEW

**Comment by Hosch250** [comment]
:+1:

File: RubberduckTests/Inspections/FunctionReturnValueNotUsedInspectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        [TestMethod]
+        public void FunctionReturnValueNotUsed_ReturnsResult_NoUsages()
+        {
+            const string inputCode =
+@"Public Function Foo(ByVal bar As String) As Integer
+    Foo = 42
+End Function
+Public Function Baz() As Integer
+End Function
+Public Sub Bar()
```
Comment (Hosch250) [inline RubberduckTests/Inspections/FunctionReturnValueNotUsedInspectionTests.cs]: Please split this into one test for each potential failure point.

File: RubberduckTests/Inspections/FunctionReturnValueNotUsedInspectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        [TestMethod]
+        public void FunctionReturnValueNotUsed_DoesNotReturnResult_WithUsages()
+        {
+            const string inputCode =
+@"Public Function Foo(ByVal bar As String) As Integer
+    Foo = 42
+End Function
+Sub Bar(ByVal fizz As Boolean)
+End Sub
+Public Sub Baz()
```
Comment (Hosch250) [inline RubberduckTests/Inspections/FunctionReturnValueNotUsedInspectionTests.cs]: Please split this as well.

File: RetailCoder.VBE/UI/RubberduckUI.resx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/RubberduckUI.resx]: Some of my changes made a conflict in this file.  Please pull from Next and manually resolve this conflict, being sure to keep my updates.

File: RetailCoder.VBE/Inspections/ConvertToProcedureQuickFix.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+﻿using Antlr4.Runtime;
```
Comment (Hosch250) [inline RetailCoder.VBE/Inspections/ConvertToProcedureQuickFix.cs]: Looks ready to merge once these issues are fixed!

**Comment by autoboosh** [comment]
Merged and fixed a merge conflict. Should work now.

**Pull request merged**